src/vcpkg/commands.z-generate-message-map.cpp

```diff
@@ -143,52 +149,74 @@ namespace vcpkg
+                {
+                    has_errors = true;
+                    msg::print(LocalizedString::from_raw(ErrorPrefix)
+                                   .append_raw(fmt::format("The message named {} starts with {}, it must be changed "
+                                                           "to prepend {} in code instead.\n",
+                                                           msg.name,
+                                                           test.prefix,
+                                                           test.prefix_name)));
+                }
             }
 
             if (Strings::contains(msg.value, "   "))
             {
                 has_errors = true;
-                msg::println_error(msgLocalizedMessageMustNotContainIndents, msg::value = msg.name);
+                msg::print(LocalizedString::from_raw(ErrorPrefix)
+                               .append_raw(fmt::format(
+                                   "The message named {} contains what appears to be indenting which must be "
+                                   "changed to use LocalizedString::append_indent instead.\n",
+                                   msg.name)));
             }
 
             if (!msg.value.empty() && msg.value.back() == '\n')
             {
                 has_errors = true;
-                msg::println_error(msgLocalizedMessageMustNotEndWithNewline, msg::value = msg.name);
+                msg::print(LocalizedString::from_raw(ErrorPrefix)
+                               .append_raw(fmt::format("The message named {} ends with a newline which should be added "
+                                                       "by formatting rather than by localization.",
+                                                       msg.name)));
             }
 
             auto mismatches = get_format_arg_mismatches(msg.value, msg.comment, format_string_parsing_error);
             if (!format_string_parsing_error.data().empty())
             {
                 has_errors = true;
-                msg::println_error(msg::format(msgGenerateMsgErrorParsingFormatArgs, msg::value = msg.name)
-                                       .append(format_string_parsing_error));
+                msg::print(LocalizedString::from_raw(ErrorPrefix)
+                               .append_raw(fmt::format("parsing format string for {}:\n", msg.name)));
             }
 
             if (!mismatches.arguments_without_comment.empty() || !mismatches.comments_without_argument.empty())
             {
                 has_errors = true;
-                msg::println_error(msgGenerateMsgIncorrectComment, msg::value = msg.name);
+                msg::print(LocalizedString::from_raw(ErrorPrefix)
+                               .append_raw(fmt::format("message {} has an incorrect comment:\n", msg.name)));
 
                 for (const auto& arg : mismatches.arguments_without_comment)
```

Comment:
Double check these -- the message seems opposite from the member name `arguments_without_comment`.